File: src/main/java/com/akagiyui/drive/entity/Announcement.java
package com.akagiyui.drive.entity;

import com.akagiyui.common.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.ToString;
import lombok.experimental.Accessors;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.DynamicInsert;


/**
 * 公告实体
 *
 * @author AkagiYui
 */
@Data
@ToString(exclude = "author")
@Accessors(chain = true)
@Entity
@Table
@DynamicInsert
public class Announcement extends BaseEntity {
    /**
     * 标题
     */
    @Column(nullable = false)
    private String title;

    /**
     * 内容
     */
    private String content;

    /**
     * 已启用
     */
    @Column(nullable = false)
    @ColumnDefault("true")
    private Boolean enabled;

    /**
     * 发布者
     */
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id")
    private User author;

}

File: app/src/main/kotlin/com/akagiyui/drive/entity/Announcement.kt
package com.akagiyui.drive.entity

import com.akagiyui.common.entity.BaseEntity
import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import org.hibernate.annotations.DynamicInsert

/**
 * 公告实体
 *
 * @author AkagiYui
 */
@Entity
@Table
@DynamicInsert
class Announcement : BaseEntity() {
    /**
     * 标题
     */
    @Column(nullable = false)
    lateinit var title: String

    /**
     * 内容
     */
    var content: String? = null

    /**
     * 已启用
     */
    @Column(nullable = false)
    @ColumnDefault("true")
    var enabled: Boolean = true

    /**
     * 发布者
     */
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id")
    lateinit var author: User
}
